src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/Event.scala

```diff
@@ -0,0 +1,212 @@
+                 br_cookies:               Option[Boolean],
+                 br_colordepth:            Option[String],
+                 br_viewwidth:             Option[Int],
+                 br_viewheight:            Option[Int],
+                 os_name:                  Option[String],
+                 os_family:                Option[String],
+                 os_manufacturer:          Option[String],
+                 os_timezone:              Option[String],
+                 dvce_type:                Option[String],
+                 dvce_ismobile:            Option[Boolean],
+                 dvce_screenwidth:         Option[Int],
+                 dvce_screenheight:        Option[Int],
+                 doc_charset:              Option[String],
+                 doc_width:                Option[Int],
+                 doc_height:               Option[Int],
+                 tr_currency:              Option[String],
+                 tr_total_base:            Option[Double],
+                 tr_tax_base:              Option[Double],
+                 tr_shipping_base:         Option[Double],
+                 ti_currency:              Option[String],
+                 ti_price_base:            Option[Double],
+                 base_currency:            Option[String],
+                 geo_timezone:             Option[String],
+                 mkt_clickid:              Option[String],
+                 mkt_network:              Option[String],
+                 etl_tags:                 Option[String],
+                 dvce_sent_tstamp:         Option[Instant],
+                 refr_domain_userid:       Option[String],
+                 refr_device_tstamp:       Option[Instant],
+                 derived_contexts:         Contexts,
+                 domain_sessionid:         Option[String],
+                 derived_tstamp:           Option[Instant],
+                 event_vendor:             Option[String],
+                 event_name:               Option[String],
+                 event_format:             Option[String],
+                 event_version:            Option[String],
+                 event_fingerprint:        Option[String],
+                 true_tstamp:              Option[Instant]) {
+
+  def inventory: Set[Data.ShreddedType] = {
+    val unstructEvent = unstruct_event
+      .data
+      .toSet
+      .map((ue: SelfDescribingData[Json]) => Data.ShreddedType(Data.UnstructEvent, ue.schema))
+
+    val derivedContexts = derived_contexts
+      .data
+      .toSet
+      .map((ctx: SelfDescribingData[Json]) => Data.ShreddedType(Data.Contexts(Data.DerivedContexts), ctx.schema))
+
+    val customContexts = contexts
+      .data
+      .toSet
+      .map((ctx: SelfDescribingData[Json]) => Data.ShreddedType(Data.Contexts(Data.CustomContexts), ctx.schema))
+
+    customContexts ++ derivedContexts ++ unstructEvent
+  }
+
+  def atomic: Map[String, Json] =
+    this.asJsonObject.toMap.flatMap {
```

Comment:
Still unnecessary traversing `Map(1, 2) - 1` is fastest possible way to delete a key from a hashmap.